src/cbapi/psc/threathunter/models.py

```diff
@@ -175,3 +177,694 @@ def children(self):
+            report.validate()
+
+    def delete(self):
+        """Deletes this feed from the ThreatHunter server.
+
+        :raise InvalidObjectError: if `id` is missing
+        """
+        if not self.id:
+            raise InvalidObjectError("missing feed ID")
+
+        self._cb.delete_object("/threathunter/feedmgr/v1/feed/{}".format(self.id))
+
+    def update(self, **kwargs):
+        """Update this feed's metadata with the given arguments.
+
+        >>> feed.update(access="private")
+
+        :param kwargs: The fields to update
+        :type kwargs: dict(str, str)
+        :raise InvalidObjectError: if `id` is missing or :py:meth:`validate` fails
+        :raise ApiError: if an invalid field is specified
+        """
+        if not self.id:
+            raise InvalidObjectError("missing feed ID")
+
+        for key, value in kwargs.items():
+            if key in self._info:
+                self._info[key] = value
+
+        self.validate()
+
+        url = "/threathunter/feedmgr/v1/feed/{}/feedinfo".format(self.id)
+        new_info = self._cb.put_object(url, self._info).json()
+        self._info.update(new_info)
+
+        return self
+
+    @property
+    def reports(self):
+        """Returns a list of :py:class:`Report` associated with this feed.
+
+        :return: a list of reports
+        :rtype: list(:py:class:`Report`)
+        """
+        return self._cb.select(Report).where(feed_id=self.id)
+
+    def replace(self, reports, append=False):
+        """Repace this feed's report with the given reports, appending
+        if requested.
+
+        :param reports: the reports to replace with
+        :type reports: list(:py:class:`Report`)
+        :param bool append: whether or not to append the given reports to the current list
+        :raise InvalidObjectError: if `id` is missing
+        """
+        if not self.id:
+            raise InvalidObjectError("missing feed ID")
+
+        rep_dicts = [report._info for report in reports]
+
+        if append:
+            rep_dicts += [report._info for report in self._reports]
+
+        body = {"reports": rep_dicts}
+
+        self._cb.post_object("/threathunter/feedmgr/v1/{}/report".format(self.id), body)
+
+
+class Report(FeedModel):
+    """Represents reports retrieved from a ThreatHunter feed.
+    """
+    urlobject = "/threathunter/feedmgr/v1/feed/{}/report"
+    primary_key = "id"
+    swagger_meta_file = "psc/threathunter/models/report.yaml"
+
+    @classmethod
+    def _query_implementation(cls, cb):
+        return ReportQuery(cls, cb)
+
+    def __init__(self, cb, model_unique_id=None, initial_data=None,
+                 feed_id=None, from_watchlist=False):
+
+        super(Report, self).__init__(cb, model_unique_id=initial_data.get("id"),
+                                     initial_data=initial_data,
+                                     force_init=False, full_doc=True)
+
+        # NOTE(ww): Warn instead of failing, since not all report operations
```

Comment:
which report operations require/don't require a feed_id? I'll file bugs and/or investigate what the rationale is behind it

Reply:
I may be answering my own question: does this have to do with whether the report is part of a watchlist or feed?

Reply:
Yep, exactly. I'll update the `NOTE` to make that more clear.